internal/tenant/query.go

```diff
@@ -0,0 +1,29 @@
+package tenant
+
+import (
+	"context"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/sourcegraph/zoekt/internal/tenant/internal/tenanttype"
+)
+
+// Match returns true if the path is a subdirectory of tenant's index directory.
+func Match(ctx context.Context, path string) bool {
+	if !EnforceTenant() {
+		return true
+	}
+
+	t, err := tenanttype.FromContext(ctx)
+	if err != nil {
+		return false
+	}
+
+	baseDir := filepath.Base(filepath.Dir(path))
```

Comment:
how user-controlled is the path here? If the path is not cleaned and absolute, this might allow to pass the check with something like `asdf/../../17/otherrepo`

Reply:
We control the path of all shards, so this should be fine.